x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerUtils.java

```diff
@@ -66,8 +69,15 @@ public static Tuple<PhysicalPlan, PhysicalPlan> breakPlanBetweenCoordinatorAndDa
             // remember the datanode subplan and wire it to a sink
             var subplan = e.child();
             dataNodePlan.set(new ExchangeSinkExec(e.source(), e.output(), e.isInBetweenAggs(), subplan));
-
-            return new ExchangeSourceExec(e.source(), e.output(), e.isInBetweenAggs());
+            List<Attribute> output;
+            if (config.pragmas().timeSeriesMode()) {
```

Comment:
This was needed because `EsTimeseriesQueryExec` is created after `AbstractPhysicalOperationProviders#groupingPhysicalOperation()` is invoked and `AggregateExec` is created.

I suspect this is the wrong place. The `_tsid` and `@timestamp` fields are needed in both the new grouping operator and the the time series source operator. For the later case `EsTimeseriesQueryExec` was added. 

I wonder whether we can define the need for these fields once in the planner? So that we don't need to indicate in two places (here and in `EsTimeseriesQueryExec`) that these fields are required.

Reply:
I think we can add these in `AbstractPhysicalOperationProviders#intermediateAttributes`.

Reply:
I see that this method is used in three places. Only in `groupingPhysicalOperation#pushableStats(...)` there is access to `LocalPhysicalOptimizerContext`, from which query pragma's are accessible. In the other cases this isn't accessible (or we need to push `EsqlConfiguration` / `LocalPhysicalOptimizerContext` down). Or we do we only care about the call to `groupingPhysicalOperation ()` in `groupingPhysicalOperation`?

Reply:
What do you think of eb00f846c1a550a4a801d4526314c7e04a6ec7cd?
Only in `Mapper#map(...)` I can't figure out whether query pragma's are used. I think I can push it down but requires a lot of plumbing.